app/src/main/java/org/mozilla/fenix/collections/SaveCollectionListAdapter.kt

```diff
@@ -78,9 +82,46 @@ class CollectionViewHolder(
     fun bind(collection: TabCollection) {
         this.collection = collection
         view.collection_item.text = collection.title
+
+        val hostNameList = collection.tabs.map { it.url.urlToTrimmedHost().capitalize() }
+
+        var tabsDisplayed = 0
+        val tabTitlesList = hostNameList.joinToString(", ") {
+            if (it.length > maxTitleLength) {
+                it.substring(
+                    0,
+                    maxTitleLength
+                ) + "..."
```

Comment:
Isn't this what android:ellipsize does using the length of the TextView instead of a hard limit on characters? I'm curious why we're reinventing the wheel here.

Reply:
We're not ellipsizing the TextView we're putting a limit on the length of the title in a list of titles in a TextView